raft-tech/TANF-app
# DAC CSV Export

## Summary of Changes
- Added support for multi-selection list filters
- Added new filters for STT code, and fiscal year
- Added support for CSV export
- Made Records read only in DAC
- Updated `Newest` filter to display the newest records associated to each STT instead of to the latest datafile.
Pull request closes #2908 

## How to Test
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up -- build
```

1. Open http://localhost:3000/ and sign in as a user with admin privileges and permissions to search indexes.
2. Submit some files as different STTs
3. View the records and verify that records for the latest datafile for the respective STTs are shown
4. Export some or all of the records as CSV
5. Verify all records are in CSV, STT has its own column, and you can freely navigate DAC while the download is occurring

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] `DIGIT team/OFA system admin/OFA admin` can export parsed data via DAC to CSV
- [x] CSV export includes field names and STT code
- [x] CSV export only includes latest parsed data for any/all STTs 
- [x] anyone with access to search_indexes cannot edit/delete parsed data via DAC
- [x] filtering capability built to abovementioned specs
   - _in-place but needs refinement_ 

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
